decoder.go

```diff
@@ -74,12 +76,15 @@ func (dec *Decoder) Decode(data []byte, pcm []int16) (int, error) {
 	if len(pcm) == 0 {
 		return 0, fmt.Errorf("opus: target buffer empty")
 	}
+	if len(pcm)%dec.channels != 0 {
+		return 0, fmt.Errorf("opus: output buffer length must be multiple of channels")
+	}
 	n := int(C.opus_decode(
 		dec.p,
 		(*C.uchar)(&data[0]),
 		C.opus_int32(len(data)),
 		(*C.opus_int16)(&pcm[0]),
-		C.int(cap(pcm)),
+		C.int(len(pcm)/dec.channels),
```

Comment:
```suggestion
		C.int(cap(pcm)/dec.channels),
```

sticking to cap for this PR, will think on cap/len later.